src/proto/uprotocol/uauthority.rs

```diff
@@ -65,4 +78,51 @@ impl UAuthority {
         self.remote = Some(Remote::Id(id));
         self
     }
+
+    /// Returns whether the `Remote` `Ip` meets conformance with IPv4 or IPv6 spec
+    ///
+    /// # Returns
+    /// Returns a `Result<IpConformance, ValidationError>` where the error means remote is not IP or there is no remote
+    /// and happy path tells us whether the IP conforms with spec or not
+    ///
+    /// # Errors
+    ///
+    /// Returns a `ValidationError` in the failure case, indicating no remote or remote is not IP
+    pub fn remote_ip_conforms(&self) -> Result<IpConformance, ValidationError> {
+        if let Some(_remote) = self.remote.as_ref() {
+            match &self.remote {
+                Some(Remote::Ip(ip)) => Ok(match ip.len() {
+                    REMOTE_IPV4_BYTES => IpConformance::IPv4,
+                    REMOTE_IPV6_BYTES => IpConformance::IPv6,
+                    _ => IpConformance::NonConformal,
+                }),
+                _ => Err(ValidationError::new("Remote is not IP")),
+            }
+        } else {
+            Err(ValidationError::new("No remote"))
+        }
+    }
+
+    /// Returns whether the `Remote` `Id` meets conformance allowable range of bytes
+    ///
+    /// # Returns
+    /// Returns a `Result<bool, ValidationError>` where the error means remote is not ID or there is no remote
+    /// and happy path tells us whether the ID conforms with allowable range of bytes or not
+    ///
+    /// # Errors
+    ///
+    /// Returns a `ValidationError` in the failure case, indicating no remote or remote is not IP
+    pub fn remote_id_conforms(&self) -> Result<bool, ValidationError> {
```

Comment:
is there really much value in returning a Result instead of just a boolean?

Reply:
i didn't tackle it in this PR, but my thought process was to allow the "bubbling up" of the `ValidationError`s that had occurred, so that a caller could e.g. append all of them together on their end.

What do you think?